test/parallel/test-shadow-realm-globals.js

```diff
@@ -0,0 +1,27 @@
+// Flags: --experimental-shadow-realm
+'use strict';
+
+require('../common');
+const { intrinsics, webIdlExposedWildcard } = require('../common/globals');
+const assert = require('assert');
+
+// Validates APIs exposed on the ShadowRealm globalThis.
+const shadowRealm = new ShadowRealm();
+const itemsStr = shadowRealm.evaluate(`
+(() => {
+  return Object.getOwnPropertyNames(globalThis).join(',');
+})();
+`);
+const items = itemsStr.split(',');
+const leaks = [];
+for (const item of items) {
+  if (intrinsics.has(item)) {
+    continue;
+  }
+  if (webIdlExposedWildcard.has(item)) {
+    continue;
+  }
+  leaks.push(item);
```

Comment:
Regarding the [CI failure](https://github.com/nodejs/node/pull/46809#issuecomment-1456198972), I think we just need to check `webIdlExposedWindow` here. It might make sense to add `nodeGlobals` to future proof the test as more APIs are added to the `ShadowRealm`.

Reply:
This is introduced in the recent change (https://github.com/nodejs/node/pull/46864). I've rebased to the top of the main branch and fixed the problem.